core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -715,12 +715,8 @@ private[spark] object Utils extends Logging {
 
   /** Get the Yarn approved local directories. */
   private def getYarnLocalDirs(conf: SparkConf): String = {
-    // Hadoop 0.23 and 2.x have different Environment variable names for the
-    // local dirs, so lets check both. We assume one of the 2 is set.
-    // LOCAL_DIRS => 2.X, YARN_LOCAL_DIRS => 0.23.X
-    val localDirs = Option(conf.getenv("YARN_LOCAL_DIRS"))
-      .getOrElse(Option(conf.getenv("LOCAL_DIRS"))
-      .getOrElse(""))
+    //YarnLocalDirs must be inside container directory. Since it will be automatically deleted when container shut downs.
+    val localDirs = Option(System.getProperty("user.dir")).getOrElse(""))
```

Comment:
So what i understood is: 
1. ExternalShuffleService is a separate process - since it must be running even when executor dies. 
2. It is per node and serves all executors of that node.
3. If one executor on node dies then the blocks will be served by another executor of that node.
4. It does not directly read files and just keep mapping of blockId to filename.

If above is correct how does it serve blocks if all the executors on particular node dies.

Am i wrong somehwere in my understanding?
--pankaj